optimizely/client/client.go

```diff
@@ -43,6 +43,29 @@ type OptimizelyClient struct {
 	executionCtx utils.ExecutionCtx
 }
 
+// Activate returns the variation that the user is bucketed into. Returns an empty string if the user is not bucketed into any variation.
+// It also sends an impression if the user is bucketed into a variation.
+func (o *OptimizelyClient) Activate(experimentKey string, userContext entities.UserContext) (string, error) {
+	decisionContext, experimentDecision, err := o.getExperimentDecision(experimentKey, userContext)
+	if experimentDecision.Variation != nil {
+		impressionEvent := event.CreateImpressionUserEvent(decisionContext.ProjectConfig, *decisionContext.Experiment, *experimentDecision.Variation, userContext)
+		o.eventProcessor.ProcessEvent(impressionEvent)
+		return experimentDecision.Variation.Key, err
+	}
+
+	return "", err
```

Comment:
are we guaranteed that err is not nil when experimentDecision.Variation == nil ?

Reply:
No, it is not guaranteed. You can have a nil variation with an error